src/full/Agda/Auto/Convert.hs

```diff
@@ -137,7 +137,7 @@ tomy imi icns typs = do
        let Datatype [con] [] = cdcont cc
        lift $ liftIO $ modifyIORef con (\cdef -> cdef {cdtype = contyp'})
 
-       projfcns <- mapM (\name -> getConst False name TMAll) (map I.unDom fields)
+       projfcns <- mapM ((\name -> getConst False name TMAll) . I.unDom) fields
```

Comment:
Wouldn't
```haskell
projfcns <- mapM (\ dom -> getConst False (I.unDom dom) TMAll) fields
```
be more readable?

Reply:
How about:
```haskell
projfcns <- mapM (flip (getConst False) TMAll . I.unDom) fields
```
Or, equivalently:
```haskell
projfcns <- mapM ((getConst False `flip` TMAll) . I.unDom) fields
```
(Such infix ``` `flip` ``` can be quite useful, but probably takes getting used to...)

Or even:
```haskell
projfcns <- mapM (($ TMAll) . getConst False . I.unDom) fields
```
:wink:

Reply:
I don't have strong views on this but, as I've said elsewhere, it is more important that the same pattern is used throughout the codebase. I'm sure that it is possible to get hlint to rewrite in any of the suggested ways.  That said, flip is not my favourite function, but can see that it's useful if you prefer a completely pointfree approach.